src/shared/exec-util.c

```diff
@@ -438,6 +438,12 @@ static int gather_environment_consume(int fd, void *arg) {
         return r;
 }
 
+const gather_stdout_callback_t gather_environment[] = {
```

Comment:
I do not see any reason to move this.

Reply:
if you touch this I'd probably add `[_STDOUT_CONSUME_MAX]` here, to fixate the size. doesn't really change anything, but given that the size is part of the API here, it's probably worth declaring it explicitly.

Reply:
> I do not see any reason to move this.

This is part of `execute_directories` API above, while functions below are about `ExecCommandFlags`.